packages/framework/aqueduct/src/data-object-factories/pureDataObjectFactory.ts

```diff
@@ -40,22 +40,31 @@ import type {
 	PureDataObject,
 } from "../data-objects/index.js";
 
+interface CreateDataObjectProps<TObj extends PureDataObject, I extends DataObjectTypes> {
+	ctor: new (props: IDataObjectProps<I>) => TObj;
+	context: IFluidDataStoreContext;
+	sharedObjectRegistry: ISharedObjectRegistry;
+	optionalProviders: FluidObjectSymbolProvider<I["OptionalProviders"]>;
+	runtimeClassArg: typeof FluidDataStoreRuntime;
+	existing: boolean;
+	initialState?: I["InitialState"];
+}
 /**
  * Proxy over PureDataObject
  * Does delayed creation & initialization of PureDataObject
  */
 async function createDataObject<
 	TObj extends PureDataObject,
 	I extends DataObjectTypes = DataObjectTypes,
->(
-	ctor: new (props: IDataObjectProps<I>) => TObj,
-	context: IFluidDataStoreContext,
-	sharedObjectRegistry: ISharedObjectRegistry,
-	optionalProviders: FluidObjectSymbolProvider<I["OptionalProviders"]>,
-	runtimeClassArg: typeof FluidDataStoreRuntime,
-	existing: boolean,
-	initProps?: I["InitialState"],
-): Promise<{
+>({
+	ctor,
+	context,
+	sharedObjectRegistry,
+	optionalProviders,
+	runtimeClassArg,
+	existing,
+	initialState: initProps,
```

Comment:
this has different names in different parts of the public api, so can't unify without breaking, and this was the least impactful place to transition between the names